core/system/src/mpu/vmpu.c

```diff
@@ -78,16 +71,23 @@ static int vmpu_sanity_checks(void)
     DPRINTF("uVisor mode: %u\n", *(__uvisor_config.mode));
 
     /* Verify the SRAM relocation. */
-    DPRINTF("uvisor_ram : @0x%08X (%u bytes) [config]\n", __uvisor_config.bss_main_start,
+    /* Note: All labels in the BSS section in the host linker script are aligned
+     * to 32 bytes. */
+    assert((uint32_t) __uvisor_config.bss_start == UVISOR_ROUND32_UP((SRAM_ORIGIN + SRAM_OFFSET)));
+
+    DPRINTF("uvisor_ram : @0x%08X (%u bytes) [config]\n",
+        __uvisor_config.bss_main_start,
         VMPU_REGION_SIZE(__uvisor_config.bss_main_start, __uvisor_config.bss_main_end));
-    DPRINTF("             (0x%08X (%u bytes) [linker]\n", SRAM_OFFSET_START, UVISOR_SRAM_LENGTH);
-    assert(__uvisor_config.bss_main_end > __uvisor_config.bss_main_start);
-    assert(VMPU_REGION_SIZE(__uvisor_config.bss_main_start, __uvisor_config.bss_main_end) == UVISOR_SRAM_LENGTH);
+    DPRINTF("             @0x%08X (%u bytes) [linker]\n",
+        SRAM_ORIGIN + SRAM_OFFSET,
+        UVISOR_SRAM_LENGTH);
+
+    /* Verify the uVisor expectations regarding its own memories. */
+    /* Note: Due to alignment, the size available to uVisor could be larger than
+     * UVISOR_SRAM_LENGTH. */
+    assert(VMPU_REGION_SIZE(__uvisor_config.bss_main_start, __uvisor_config.bss_main_end) >= UVISOR_SRAM_LENGTH);
     assert(&__stack_end__ <= __uvisor_config.bss_main_end);
 
-    assert((uint32_t) __uvisor_config.bss_main_start == SRAM_OFFSET_START);
```

Comment:
This assertion is still valid, right? We could leave it in (and not remove it in this commit).